index.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    i = 0
+    while i < list.length
+      yield(list[i])
+      i += 1
+    end
+    list
+  end
+
+  def my_each_with_index()
+    return enum_for unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    index = 0
+    while index < list.length
+      yield(self[index], index)
+      index += 1
+    end
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    filter = []
+    my_each { |element| filter.push(element) if yield(element) }
+    filter
+  end
+
+  def my_all?(*args)
+    lever = true
+    if !args[0].nil?
+      my_each { |element| lever = false unless args[0] == element }
+    elsif !block_given?
+      my_each { |element| lever = false unless element }
+    else
+      my_each { |element| lever = false unless yield(element) }
+    end
+    lever
+  end
```

Comment:
* [ ] should have working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_all? 

- [ ] when a class is passed as an argument returns true only if all of the collection is a member of such class, otherwise, it should return false

```ruby     
[1, 3, 9].my_all?(Integer) == [1, 3, 9].all?(Integer) => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/79289944-02738a80-7e88-11ea-98f7-d7a25ff6ad30.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/79289968-11f2d380-7e88-11ea-8ad5-d9ca49df03e6.png)

- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex

```ruby     
%w[dog door rod blade].my_all?(/d/) == %w[dog door rod blade].all?(/d/) => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/79290025-32229280-7e88-11ea-9c18-2dd13961bf15.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688741-31bc6900-6f66-11ea-90ec-b6a7cee7e6ca.png)

Reply:
solved